ql/src/java/org/apache/hadoop/hive/ql/optimizer/ColumnPrunerProcCtx.java

```diff
@@ -239,6 +275,169 @@ public ParseContext getParseContext() {
   }
 
   /**
+   * Creates the list of internal group paths for select * expressions.
+   *
+   * @param op        The select operator.
+   * @param paths The list of nested column paths returned by the children of the
+   *                  select operator.
+   * @return List<String> of the nested column path from leaf to the root.
+   */
+  public List<String> getSelectNestedColPathsFromChildren(
+    SelectOperator op,
+    List<String> paths) {
+    List<String> groups = new ArrayList<>();
+    SelectDesc conf = op.getConf();
+
+    if (paths != null  && conf.isSelStarNoCompute()) {
+      groups.addAll(paths);
+      return groups;
+    }
+
+    List<ExprNodeDesc> selectDescs = conf.getColList();
+
+    List<String> outputColumnNames = conf.getOutputColumnNames();
+    for (int i = 0; i < outputColumnNames.size(); i++) {
+      if (paths == null || paths.contains(outputColumnNames.get(i))) {
+        ExprNodeDesc desc = selectDescs.get(i);
+        List<String> gp = getNestedColPathByDesc(desc);
+        if (!gp.isEmpty()) {
```

Comment:
I know it's "to avoid adding empty set", what I mean is this check may not be necessary.
You can just have `groups.addAll(gp);`, right? to make it simpler.